libnymea/hardware/modbus/modbusrtuhardwareresource.h

```diff
@@ -0,0 +1,59 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+*
+* Copyright 2013 - 2021, nymea GmbH
+* Contact: contact@nymea.io
+*
+* This file is part of nymea.
+* This project including source code and documentation is protected by
+* copyright law, and remains the property of nymea GmbH. All rights, including
+* reproduction, publication, editing and translation, are reserved. The use of
+* this project is subject to the terms of a license agreement to be concluded
+* with nymea GmbH in accordance with the terms of use of nymea GmbH, available
+* under https://nymea.io/license
+*
+* GNU Lesser General Public License Usage
+* Alternatively, this project may be redistributed and/or modified under the
+* terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; version 3. This project is distributed in the hope that
+* it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this project. If not, see <https://www.gnu.org/licenses/>.
+*
+* For any further details and any questions please contact us under
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#ifndef MODBUSRTUHARDWARERESOURCE_H
+#define MODBUSRTUHARDWARERESOURCE_H
+
+#include <QList>
+#include <QObject>
+
+#include "hardwareresource.h"
+#include "modbusrtumaster.h"
+
+class ModbusRtuHardwareResource : public HardwareResource
+{
+    Q_OBJECT
+public:
+    virtual QList<ModbusRtuMaster *> modbusRtuMasters() const = 0;
```

Comment:
Would this allow one plugin to break another plugins object?

Can you think of a way where a plugin can only obtain its own objects? Similar as we did with zigbee

Reply:
The entire point of this is to give different plugins access to one RTU master...each plugin can send and receive data from the master, working on it's own registers and addresses on a modbus layer. The RTU master is just the "transport" which can be shared with any plugin. What a plugin does on modbus layer is a different story, but that is not business of the transport.